crates/next-custom-transforms/src/transforms/lint_codemod_comments.rs

```diff
@@ -0,0 +1,82 @@
+use swc_core::{
+    common::{
+        comments::{Comment, Comments},
+        errors::HANDLER,
+    },
+    ecma::{utils::swc_common::Span, visit::Visit},
+};
+
+struct LintErrorComment<C>
+where
+    C: Comments,
+{
+    comments: C,
+}
+
+pub fn lint_codemod_comments<C>(comments: C) -> impl Visit
+where
+    C: Comments,
+{
+    LintErrorComment { comments }
+}
+
+// declare a const of comment prefix
+const COMMENT_ERROR_PREFIX: &str = "@next-codemod-error";
+const COMMENT_BYPASS_PREFIX: &str = "@next-codemod-ignore";
+
+impl<C> LintErrorComment<C>
+where
+    C: Comments,
+{
+    fn lint(&self, comment: &Comment, is_leading: bool) {
+        let trimmed_text = comment.text.trim();
+        // if comment contains @next/codemod comment "@next-codemod-error",
+        // report an error from the linter to fail the build
+        if trimmed_text.contains(COMMENT_ERROR_PREFIX) {
+            let span = if is_leading {
+                comment
+                    .span
+                    .with_lo(comment.span.lo() - swc_core::common::BytePos(1))
+            } else {
+                comment
+                    .span
+                    .with_hi(comment.span.hi() + swc_core::common::BytePos(1))
+            };
+            let action = trimmed_text.replace(COMMENT_ERROR_PREFIX, "");
+            let err_message = format!(
+                "You have unresolved @next/codemod comment \"{}\" that need to be reviewed, \
+                 please address and remove them to proceed with the build.\nYou can also bypass \
+                 the build error by replacing \"{}\" with \"{}\".",
```

Comment:
```suggestion
                "You have an unresolved @next/codemod comment \"{}\" that needs review.\nAfter review, either remove the comment if you made the necessary changes or replace \"{}\" with \"{}\" to bypass the build error if no action at this line can be taken.",
```

Focus more on what you want to do under which circumstance.